go/obscuronode/enclave/state.go

```diff
@@ -118,6 +118,33 @@ func updateState(b *types.Block, s Storage, blockResolver BlockResolver) *blockS
 		return nil
 	}
 
+	// Detect if the incoming block contains the genesis rollup, and generate an updated state.
+	// Handle the case of the block containing the genesis being processed multiple times.
+	genesisState, isGenesis := handleGenesisRollup(b, s, rollups, genesisRollup)
+	if isGenesis {
+		return genesisState
+	}
+
+	// To calculate the state after the current block, we need the state after the parent.
+	// If this point is reached, there is a parent state guaranteed, because the genesis is handled above
+	parentState, parentFound := s.FetchBlockState(b.ParentHash())
+	if !parentFound {
+		// go back and calculate the State of the Parent
+		p, f := s.FetchBlock(b.ParentHash())
+		if !f {
+			panic("Could not find block parent. This should not happen.")
+		}
+		parentState = updateState(p, s, blockResolver)
+	}
+
+	bs := calculateBlockState(b, parentState, s, blockResolver, rollups)
+
+	s.SetBlockState(b.Hash(), bs)
+
+	return bs
+}
+
+func handleGenesisRollup(b *types.Block, s Storage, rollups []*Rollup, genesisRollup *Rollup) (genesisState *blockState, isGenesis bool) {
 	// the incoming block holds the genesis rollup
 	// todo change this to an hardcoded hash on testnet/mainnet
```

Comment:
```suggestion
	// the incoming block holds the genesis rollup
	// calculate and return the new block state
	// todo change this to an hardcoded hash on testnet/mainnet
```

Reply:
didn't know you could suggest changes. this is cool